### PR TITLE
[FEATURE] #97 title and canonical field names can be set over typoscript

### PR DESCRIPTION
**Pull request by sfsmfc**
... and update condition for disable plugin in frontend

### CODE REVIEW

File: Classes/ViewHelpers/KeyValueViewHelper.php [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+use TYPO3\CMS\Fluid\Core\ViewHelper\AbstractViewHelper;
+
+/**
+ * Class KeyValueViewHelper
+ *
+ * @package YoastSeoForTypo3\YoastSeo\ViewHelpers
+ */
+class KeyValueViewHelper extends AbstractViewHelper
+{
+
```
Comment (haassie) [inline Classes/ViewHelpers/KeyValueViewHelper.php]: these empty lines causes travis to fail.

File: Classes/Controller/ModuleController.php [hunk 2]
```diff
@@ -270,13 +270,13 @@ public function saveAction()
         $languageId = (int)$this->request->getArgument('language');
 
         $fields = array();
-        $this->addFieldToArray($fields, 'tx_yoastseo_title', 'snippet-editor-title');
+        $this->addFieldToArray($fields, $this->settings['titleFieldName'], 'snippet-editor-title');
         $this->addFieldToArray($fields, 'description', 'snippet-editor-meta-description');
```
Comment (haassie) [inline Classes/Controller/ModuleController.php]: Maybe it is good to also make the description field configurable

**Comment by haassie** [comment]
Doesn't you miss the change in the template of the metadata? And maybe it is possible to integrate #92 with this? If you don't have time, I can fix this for you if you want.

**Comment by sfsmfc** [comment]
Hi @haassie,

the template changes can be simple down, if somebody overrides the template in its own sitepackage. Therefore I think, it should not be done automatically. We can do that, but we need to use the new keyvalue viewhelper for this. If you think it is neccessary, I will do this.

Regards

Simon

**Comment by haassie** [comment]
After some hours of sleep, I think we should make it as easy as possible. So I was thinking about using the TypoScript libs that are used for the preview, also in the template. How do you think about that @sfsmfc ? In that way the real metatag in frontend is using the same title as the snippet preview and you only have to configure it once. For the editing part we can use the viewhelper you introduced indeed.

File: Classes/ViewHelpers/KeyValueViewHelper.php (hunk 1 above)
Comment (haassie) [reply Classes/ViewHelpers/KeyValueViewHelper.php]: @sfsmfc line 12 is causing the fail of travis. No empty line after the opening brace of a class.

**Comment by haassie** [comment]
Some small changes in PR. Done in https://github.com/Yoast/Yoast-SEO-for-TYPO3/pull/102

**Pull request closed without merging**